From 44f16d744267202f16e57f1241407b6144b778f2 Mon Sep 17 00:00:00 2001
Message-Id: <patch-14.19-44f16d74426-20210331T190531Z-avarab@gmail.com>
In-Reply-To: <cover-00.19-00000000000-20210331T190531Z-avarab@gmail.com>
References: <87o8fcqrg8.fsf@evledraar.gmail.com>
	<cover-00.19-00000000000-20210331T190531Z-avarab@gmail.com>
From: =?UTF-8?q?=C3=86var=20Arnfj=C3=B6r=C3=B0=20Bjarmason?=
 <avarab@gmail.com>
Date: Wed, 31 Mar 2021 15:34:09 +0200
Subject: [PATCH v5 14/18] tree-walk.h API doc: improve documentation of
 get_tree_entry()
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Change a mention of sha1 to OID and change the comment to a listing of
functions discussed below, right now there's only one function, but
subsequent commits will add more.

Signed-off-by: Ævar Arnfjörð Bjarmason <avarab@gmail.com>
---
 tree-walk.h | 8 +++++---
 1 file changed, 5 insertions(+), 3 deletions(-)

diff --git a/tree-walk.h b/tree-walk.h
index 09e40d9221d..cd8da84f56c 100644
--- a/tree-walk.h
+++ b/tree-walk.h
@@ -166,11 +166,13 @@ struct traverse_info {
 };
 
 /**
- * Find an entry in a tree given a pathname and the sha1 of a tree to
+ * Find an entry in a tree given a pathname and the OID of a tree to
  * search. Returns 0 if the entry is found and -1 otherwise.
  *
- * The third and fourth parameters are set to the entry's sha1 and
- * mode respectively.
+ * You always need a pointer to an appropriate variable to fill in
+ * (NULL won't do!). That variable is:
+ *
+ * get_tree_entry(): unsigned short mode
  */
 int get_tree_entry(struct repository *, const struct object_id *, const char *,
 		   struct object_id *,
-- 
2.31.1.474.g72d45d12706

